include/Symbol.h

```diff
@@ -270,27 +271,29 @@ class ProceduralBlock : public Symbol {
         : children(children), kind(kind), scope(scope) {}
 };
 
-class FormalArgumentSymbol : public Symbol {
+class FormalArgumentSymbol : public VariableSymbol {
```

Comment:
Yeah, possibly we could add another base type for both, but I think an argument can actually have some of the modifiers (such as const and var).